File: 백준/Bronze/6679. 싱기한 네자리 숫자/싱기한 네자리 숫자.cc
#include <bits/stdc++.h>

using namespace std;

int a, b, c;
int sum1, sum2, sum3;
int idx, N;

int main(void){
    ios::sync_with_stdio(0); cin.tie(0);
    
    for( int i = 2992; i <= 9999 ; i++ ){
        N = i;
        while( N > 0 ){
            a = N % 10;
            N /= 10;
            sum1 += a;
        }
        N = i;
        while( N > 0 ){
            b = N % 12;
            N /= 12;
            sum2 += b;
        }
        N = i;
        while( N > 0 ){
            c = N % 16;
            N /= 16;
            sum3 += c;
        }
        if( sum1 == sum2 && sum2 == sum3 ){
            cout << i << '\n';
        }
        sum1 = sum2 = sum3 = 0;
    }
    
    return 0;
}

File: 백준/Bronze/6679. 싱기한 네자리 숫자/싱기한 네자리 숫자.py
for i in range(2992, 10000):
    N = i
    sum1 = 0
    sum2 = 0
    sum3 = 0

    temp = N
    while temp > 0:
        a = temp % 10
        temp //= 10
        sum1 += a

    temp = N
    while temp > 0:
        b = temp % 12
        temp //= 12
        sum2 += b

    temp = N
    while temp > 0:
        c = temp % 16
        temp //= 16
        sum3 += c

    if sum1 == sum2 and sum2 == sum3:
        print(i)
